pkg/solace/endpoint_provisioner.go

```diff
@@ -0,0 +1,135 @@
+	// turns the "no such queue" error into nil.
+	// Returns immediately and registers a callback that will receive an
+	// error if deprovision on the broker fails.
+	DeprovisionAsyncWithCallback(queueName string, ignoreMissing bool, callback func(err error))
+
+	// FromConfigurationProvider sets the configuration based on the specified configuration provider.
+	// The following are built in configuration providers:
+	// - EndpointPropertyMap - This can be used to set an EndpointProperty to a value programatically.
+	//
+	// The EndpointPropertiesConfigurationProvider interface can also be implemented by a type
+	// to have it act as a configuration factory by implementing the following:
+	//
+	//   func (type MyType) GetConfiguration() EndpointPropertyMap {...}
+	//
+	// Any properties provided by the configuration provider are layered over top of any
+	// previously set properties, including those set by specifying various strategies.
+	// Can be used to clone a EndpointProvisioner object.
+	FromConfigurationProvider(properties config.EndpointPropertiesConfigurationProvider) EndpointProvisioner
+
+	// Returns a copy of the current configuration held.
+	GetConfiguration() config.EndpointPropertyMap
+
+	// WithProperty will set an individual queue property by name. Does not perform type checking.
+	WithProperty(propertyName config.EndpointProperty, propertyValue interface{}) EndpointProvisioner
+
+	// WithDurability will set the durability property for the endpoint.
+	// True for durable, false for non-durable.
+	WithDurability(durable bool) EndpointProvisioner
+
+	// WithExclusiveAccess will set the endpoint access type.
+	// True for exclusive, false for non-exclusive.
+	WithExclusiveAccess(exclusive bool) EndpointProvisioner
+
+	// WithDiscardNotification will set the notification behaviour on message discards.
+	// True to notify senders about discards, false not to.
+	WithDiscardNotification(notifySender bool) EndpointProvisioner
+
+	// WithMaxMessageRedelivery will sets the number of times messages from the
+	// queue will be redelivered before being diverted to the DMQ.
+	WithMaxMessageRedelivery(count uint) EndpointProvisioner
+
+	// WithMaxMessageSize will set the maximum message size in bytes the queue will accept.
+	WithMaxMessageSize(count uint) EndpointProvisioner
+
+	// WithPermission will set the queue's permission level for others.
+	// The levels are supersets of each other, can not be combined and the last one set will take effect.
+	WithPermission(permission config.EndpointPermission) EndpointProvisioner
+
+	// WithQuotaMB will set the overall size limit of the queue in MegaBytes.
+	WithQuotaMB(quota uint) EndpointProvisioner
+
+	// WithTTLPolicy will set how the queue will handle the TTL value in messages.
+	// True to respect it, false to ignore it.
+	WithTTLPolicy(respect bool) EndpointProvisioner
+}
+
+// ProvisionOutcome - the EndpointProvisioner.Provision and EndpointProvisioner.Deprovision operations
```

Comment:
Comment mentions that this outcome is used in the prov and deprov operations but we are only returning an `error` from deprovision.  The arch agrees with your implementation not this comment but after writing some code with this I'd prefer they both return the same type (or channel of the same type) 
@cjwmorgan-sol thoughts here? Not sure if this has big impacts on other areas of deprovision is we change it form just an error

Reply:
Thanks, I have updated the comment @sdewilde